lib/max_subarray.py

```diff
@@ -2,11 +2,21 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n), iterates through relative to the size of the list given.
+        Space Complexity: ? O(1), we will only ever make 2 
```

Comment:
✨ Notice how better time complexity this approach achieves over a "naïve" approach of checking for the maximum achievable sum starting from every position and every length. The correctness of this approach might not be apparent, so I definitely encourage reading a bit more about it. [This](https://towardsdatascience.com/maximum-subarray-problem-and-kadanes-algorithm-4cb1ce91be72#8dc2) has a fairly good explanation, as well as a description of why this is considered a dynamic programming approach (on the face it might not "feel" like one).

Since like the fibonacci sequence, we are able to maintain a sliding window of recent values to complete our calculation, we can do it with a constant O(1) amount of storage.